### PR TITLE
Make filename endings case insensitive.

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

**Comment by davidgiven** [comment]
Thanks! I left a comment...

**Comment by tdaede** [comment]
I don't see a comment, did you submit the review?

**Comment by davidgiven** [comment]
Ah, I didn't know I had to. Not a fan of the github UI.

File: lib/utils.cc [hunk 1]
```diff
@@ -7,11 +7,15 @@ bool beginsWith(const std::string& value, const std::string& ending)
...
 }
 
+// Case-insensitive for endings within ASCII.
 bool endsWith(const std::string& value, const std::string& ending)
 {
     if (ending.size() > value.size())
         return false;
-    return std::equal(ending.rbegin(), ending.rend(), value.rbegin());
+    std::string lowercase;
+    std::transform(value.rbegin(), value.rbegin() + ending.size(), lowercase.begin(), [](unsigned char c){ return std::tolower(c); });
```
[the remark this answers is not in the document]
Comment (tdaede) [reply lib/utils.cc:16]: Not sure, I updated it to be initialized just in case.

[the remark this answers is not in the document]
Comment (davidgiven) [reply lib/utils.cc:16]: It's things like this that make me not a fan of the STL.

[the remark this answers is not in the document]
Comment (tdaede) [reply lib/utils.cc:16]: heh yeah, I was wishing fore Rust's iterators while writing this.

**Pull request merged**